hack/build/bazel-build-builder.sh

```diff
@@ -28,15 +28,29 @@ if [ "${CDI_CONTAINER_BUILDCMD}" = "buildah" ]; then
     fi
 fi
 
-if ! git diff-index --quiet HEAD~1 hack/build/docker; then
-    #Since this only runs during the post-submit job, the PR will have squashed into a single
-    #commit and we can use HEAD~1 to compare.
+# When this runs during the post-submit job, the PR will have squashed into a single
+# commit and we can use HEAD~1 to compare. The exit code of the git diff will therefore
+# be 0 in this case, so negating it indicates that yes this is a post-submit job and
+# we should re-build the builder. Separating out this logic from the test for clarity
+
+git diff-index --quiet HEAD~1 hack/build/docker
+POST_SUBMIT=$((1-$?))
+
+# The other circumstance in which we need to build the builder image is
+# in the course of test and development of the builder image itself. 
+# we'll signal this circumstance by setting the env variable ADHOC_BUILDER
+#
+# also instead of hard-coding the UNTAGGED_BUILDER_IMAGE, we're going
+# to use DOCKER_PREFIX as it is set in config.sh and used elsewhere in 
+# the build system, to introduce a little more consistency  
+
+if ((POST_SUBMIT)) || [ x"${ADHOC_BUILDER}" != "x" ] ; then
```

Comment:
Why not use `-eq` and `-z` or `-n` so something like
```bash
git diff-index --quiet HEAD~1 hack/build/docker

if [[ $? -eq 0 ]] || [[ -n ${ADHOC_BUILDER} ]]; then
...
fi
```

Reply:
@awels if you feel strongly about this feel free to hold the PR

Reply:
I don't feel super strongly about it. It is not like this is run a ton of times and requires optimization. We are creating subshells with the `$((1-$?))` where a simple number comparison will do.